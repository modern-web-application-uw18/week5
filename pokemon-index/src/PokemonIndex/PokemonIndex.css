.PokemonIndex {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}
.PokemonIndexItem {
    border: black solid 1px;
    margin: 1px;
    margin-bottom: 3px;
    width: 40%;
    display: flex;
    flex-direction: row;
    align-items: center;
    
}
.PokemonIndexDetails {
    display: flex;
    flex-direction: column;
    align-items: center;
    flex-grow: 1;

}

.PokemonLink {
    font-size: 20px;
    font-family: Monaco;
    text-decoration: none;
}

.PokemonIndexImage {
    border-right: grey solid 1px;
}