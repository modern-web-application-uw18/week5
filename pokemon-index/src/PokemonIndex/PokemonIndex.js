import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './PokemonIndex.css';
class PokemonIndex extends Component {
    constructor(props) {
      super(props);
      this.state = {
        pokemons: [],
        pageOffset: 0 
      }
    }
  
    componentWillMount() {
      this.getPokemonIndex();
      
    } 

    getPokemonIndex = () => {
      fetch(`https://pokeapi.co/api/v2/pokemon/?limit=20&offset=${this.state.pageOffset}`)
      .then(response => response.json())
      .then(data => {
        this.setState((prevState, props) => {
          return {
            pokemons: data.results
          }
        })
      })
      .catch(error => console.log(error));
    }

    next = () => {
      this.setState(
        (prevState) => {
          return {
            pageOffset: prevState.pageOffset + 20
          }
        },
        this.getPokemonIndex
      )
    }

    previous = () => {
      this.setState(
        (prevState) => {
          return {
            pageOffset: (prevState.pageOffset - 20 < 0 ? 0 : prevState.pageOffset - 20)
          }
        },
        this.getPokemonIndex
      )
    }

    render() {
      const pokemonIndex = this.state.pokemons.map((pokemon,idx) => {
        let pokemonId = pokemon.url.split('/')[6];
        let pokemonURL = "/pokemon/"+pokemonId;
        let spriteURL = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/"+pokemonId+".png"
        return(
          <div className="PokemonIndexItem" key={idx}>
              <img className="PokemonIndexImage" src={spriteURL} alt="Pokemon Sprite"/>
              <div className="PokemonIndexDetails">
                <Link className = "PokemonLink" to={pokemonURL}>{pokemon.name}</Link>
              </div>
          </div>
        )
      })
  
      return(
        <div>
          <h1>Pokemon Index</h1>
          <p>Click on the Pokemon name to learn more about that Pokemon</p>
          <div className="PokemonIndex">{pokemonIndex}</div>
          <button onClick={this.previous}>Previous</button>
          <button onClick={this.next}>Next</button>
        </div>
      )
    }
  }

  export default PokemonIndex;