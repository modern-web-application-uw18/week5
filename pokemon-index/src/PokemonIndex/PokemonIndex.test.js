import React from 'react';
import ReactDOM from 'react-dom';
import PokemonIndex from './PokemonIndex.js';
import { MemoryRouter as Router, Route } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    const component = (
        <Router>
            <Route exact path="/" component={PokemonIndex} />
        </Router>
    );
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render a index page', () => {
    const div = document.createElement('div');
    const component = TestRenderer.create(
        <Router>
            <Route exact path="/" component={PokemonIndex} />
        </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
});