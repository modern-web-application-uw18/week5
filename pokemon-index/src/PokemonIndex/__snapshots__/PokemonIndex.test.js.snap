// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a index page 1`] = `
<div>
  <h1>
    Pokemon Index
  </h1>
  <p>
    Click on the Pokemon name to learn more about that Pokemon
  </p>
  <div
    className="PokemonIndex"
  />
  <button
    onClick={[Function]}
  >
    Previous
  </button>
  <button
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
